colabs/wandb-artifacts/Artifact_Basics.ipynb

```diff
@@ -0,0 +1,517 @@
+      "metadata": {
+        "id": "f-RG3qgFmAH8"
+      },
+      "outputs": [],
+      "source": [
+        "import pandas\n",
+        "df = pandas.read_csv(f\"{directory}/file1.csv\")\n",
+        "# subsample to 50% of the original size\n",
+        "df_subsampled = df.sample(frac=0.5, random_state=1)\n",
+        "# save the subsampled dataframe to a new file.\n",
+        "df_subsampled.to_csv(f\"{directory}/file1.csv\", index=False)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ceBT3YXmmAH8"
+      },
+      "source": [
+        "Now we have a new subsampled version of our dataset locally, we can log the new version to Weights & Biases."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "4vDf1YL7mAH8"
+      },
+      "outputs": [],
+      "source": [
+        "run = wandb.init(project=\"artifact-basics\")\n",
+        "run.log_artifact(artifact_or_path=f\"{directory}/file1.csv\", name=\"my_first_artifact\", type=\"dataset\", aliases =[\"subsampled\"])\n",
+        "run.finish()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0ZO4QnyrmAH8"
+      },
+      "source": [
+        "Now the sampled dataset will be logged to the `my_first_artifact` Artifact as a new version.\n",
+        "\n",
+        "The Artifact has also been given a custom `alias`, which is a unique label for this Artifact version. While the `alias` is currently `subsampled`, the default aliases is `vN`, where `N` is the number of versions the Artifact has. This increments automatically. You can always access specific versions of an Artifact by using an alias."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "AyANijPlIN5m"
+      },
+      "source": [
+        "# Update Artifact version metadata"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "e4psfYzGeAVS"
+      },
+      "source": [
+        "You can update the `description`, `metadata`, and `alias` of an artifact on the W&B platform during or outside a W&B Run.\n",
+        "\n",
+        "\n",
+        "This example changes the `description` of the `my_first_artifact` artifact inside a run:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "16fCvZUZJD29"
+      },
+      "outputs": [],
+      "source": [
+        "run = wandb.init(project=\"artifact-basics\")\n",
+        "artifact = run.use_artifact(artifact_or_name=\"my_first_artifact:subsampled\")\n",
+        "artifact.description = \"This is an edited description.\"\n",
+        "artifact.metadata = {\"source\": \"local disk\", \"internal data owner\": \"platform team\"}\n",
+        "artifact.save()  # persists changes to an Artifact's properties\n",
+        "run.finish()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
```

Comment:
One last nit: let's remove code block since it doesn't show how to do use it...we leave it as an exercise (but we don't have an example to show if they are right or wrong).